src/Lumper/BSP/Lumps/BspLumps/PakFileLump.cs

```diff
@@ -2,120 +2,180 @@
+                Zip.AddEntry(entry.Key, stream, closeStream);
+            }
         }
 
-        public void SetZipArchive(ZipArchive zip, bool compress)
+        private Stream SaveZip()
         {
-            var temp = new MemoryStream();
-            var compressionType = compress
-                                    ? SharpCompress.Common.CompressionType.LZMA
-                                    : SharpCompress.Common.CompressionType.None;
-            zip.SaveTo(temp, new SharpCompress.Writers.WriterOptions(compressionType));
-            DataStream = temp;
+            var dataStream = new MemoryStream();
+            var compressionType = Compress
+                                    ? CompressionType.LZMA
+                                    : CompressionType.None;
+            Zip.SaveTo(dataStream, new WriterOptions(compressionType));
+            return dataStream;
         }
-    }
 
-    public class ZipJsonConverter : JsonConverter<ZipArchive>
-    {
-        public override void WriteJson(JsonWriter writer,
-                                       ZipArchive? value,
-                                       JsonSerializer serializer)
+        public override void Write(Stream stream)
         {
-            if (value is null)
-                return;
-            var zip = value;
-            writer.WriteStartArray();
-            foreach (var entry in zip.Entries)
+            try
             {
-                var md5 = MD5.Create().ComputeHash(entry.OpenEntryStream());
-                JObject o = JObject.FromObject(new
-                {
-                    entry.Key,
-                    MD5 = md5
-                });
-                o.WriteTo(writer);
+
+                UpdateZip(true);
+                using var dataStream = SaveZip();
+                dataStream.Seek(0, SeekOrigin.Begin);
+                dataStream.CopyTo(stream);
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
             }
-            writer.WriteEndArray();
         }
 
-        public override ZipArchive? ReadJson(JsonReader reader,
-                                             Type objectType,
-                                             ZipArchive? existingValue,
-                                             bool hasExistingValue,
-                                             JsonSerializer serializer)
+        public override bool Empty()
         {
-            throw new NotImplementedException(
-                "Unnecessary because CanRead is false. " +
-                "The type will skip the converter.");
+            return !Entries.Any();
         }
 
-        public override bool CanRead
+        public PakFileLump(BspFile parent) : base(parent)
         {
-            get { return false; }
+            Compress = false;
         }
     }
 
+    /*
+        public class ZipJsonConverter : JsonConverter<ZipArchive>
```

Comment:
Huh? If this is being left out temporarily best to leave a comment, otherwise delete. Really don't like having big blocks of random code sitting around.